art/estimators/classification/tensorflow.py

```diff
@@ -910,6 +910,34 @@ def class_gradient(self, x, label=None, **kwargs):
 
         return gradients
 
+    def loss_gradient_framework(self, x, y, **kwargs):
+        """
+        Compute the gradient of the loss function w.r.t. `x`.
+
+        :param x: Input with shape as expected by the model.
+        :type x: `Tensor`
+        :param y: Target values (class labels) one-hot-encoded of shape (nb_samples, nb_classes) or indices of shape
+                  (nb_samples,).
+        :type y: `np.ndarray`
+        :return: Gradients of the same shape as `x`.
+        :rtype: `Tensor`
+        """
+        import tensorflow as tf
+
```

Comment:
I think in the framework-specific implementations we are currently not applying preprocessing and preprocessor defences. I think we should add checks somewhere to warn the user, for example if the provided classifier has preprocessing and preprocessor defences defined.

Reply:
Added the warning in the init func of each framework specific implementation.